apps/fipsinstall.c

```diff
@@ -0,0 +1,441 @@
+    }
+    buf2 = OPENSSL_hexstr2buf(s, &len);
+    if (buf2 == NULL
+            || (size_t)len != install_mac_len
+            || memcmp(install_mac, buf2, install_mac_len) != 0) {
+        BIO_printf(bio_err, "Install indicator status mismatch\n");
+        goto end;
+    }
+    ret = 1;
+end:
+    OPENSSL_free(buf1);
+    OPENSSL_free(buf2);
+    NCONF_free(conf);
+    return ret;
+}
+
+int fipsinstall_main(int argc, char **argv)
+{
+    int ret = 1, verify = 0;
+    BIO *module_bio = NULL, *mem_bio = NULL, *fout = NULL;
+    char *in_fname = NULL, *out_fname = NULL, *prog, *section_name = NULL;
+    char *prov_name = NULL, *module_fname = NULL;
+    static const char *mac_name = DEFAULT_MAC_NAME;
+    EVP_MAC_CTX *ctx = NULL, *ctx2 = NULL;
+    STACK_OF(OPENSSL_STRING) *opts = NULL;
+    OPTION_CHOICE o;
+    unsigned char *read_buffer = NULL;
+    unsigned char module_mac[EVP_MAX_MD_SIZE];
+    size_t module_mac_len = EVP_MAX_MD_SIZE;
+    unsigned char install_mac[EVP_MAX_MD_SIZE];
+    size_t install_mac_len = EVP_MAX_MD_SIZE;
+    EVP_MAC *mac = NULL;
+    CONF *conf = NULL;
+    OPENSSL_CTX *lib_ctx = NULL;
+    OSSL_PROVIDER *default_prov = NULL;
+
+    section_name = DEFAULT_FIPS_SECTION;
+
+    prog = opt_init(argc, argv, fipsinstall_options);
+    while ((o = opt_next()) != OPT_EOF) {
+        switch (o) {
+        case OPT_EOF:
+        case OPT_ERR:
+opthelp:
+            BIO_printf(bio_err, "%s: Use -help for summary.\n", prog);
+            goto end;
+        case OPT_HELP:
+            opt_help(fipsinstall_options);
+            ret = 0;
+            goto end;
+        case OPT_IN:
+            in_fname = opt_arg();
+            break;
+        case OPT_OUT:
+            out_fname = opt_arg();
+            break;
+        case OPT_PROV_NAME:
+            prov_name = opt_arg();
+            break;
+        case OPT_MODULE:
+            module_fname = opt_arg();
+            break;
+        case OPT_SECTION_NAME:
+            section_name = opt_arg();
+            break;
+        case OPT_MAC_NAME:
+            mac_name = opt_arg();
+            break;
+        case OPT_MACOPT:
+            if (opts == NULL)
+                opts = sk_OPENSSL_STRING_new_null();
+            if (opts == NULL || !sk_OPENSSL_STRING_push(opts, opt_arg()))
+                goto opthelp;
+            break;
+        case OPT_VERIFY:
+            verify = 1;
+            break;
+        }
+    }
+    argc = opt_num_rest();
+    if (module_fname == NULL
+        || (verify && in_fname == NULL)
+        || (!verify && (out_fname == NULL || prov_name == NULL))
+        || opts == NULL
+        || argc != 0)
+        goto opthelp;
+
+    module_bio = bio_open_default(module_fname, 'r', FORMAT_BINARY);
+    if (module_bio == NULL) {
+        BIO_printf(bio_err, "Failed to open module file\n");
+        goto end;
+    }
+
+    lib_ctx = OPENSSL_CTX_new();
+    if (lib_ctx == NULL)
+        goto end;
+    default_prov = OSSL_PROVIDER_load(lib_ctx, "default");
```

Comment:
Why?

Reply:
Err this is probably a leftover (originally it was needed). I can delete..

Reply:
Please do